File: P1920.py
def binarySearch(x):
    low, high = 0, len(list_N)-1
    mid = (low+high)//2
    while low <= high:
        if x == list_N[mid]:
            return mid
        elif x < list_N[mid]:
            high = mid-1
            mid = (low+high)//2
        else:
            low = mid+1
            mid = (low+high)//2
    return -1

N = int(input())
list_N = list(map(int, input().split()))
list_N.sort()
M = int(input())
list_M = list(map(int, input().split()))

for i in list_M:
    if binarySearch(i) == -1:
        print(0)
    else:
        print(1)

File: P1920.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> v1; 
vector<int> v2;

int binarySearch(int x){
    int low= 0, high = v1.size()-1;
    while (low<=high){
        int mid = (low+high)/2;
        if(x == v1[mid]) return mid;
        else if(x < v1[mid]) high = mid-1;
        else low = mid+1;
    }
    return -1;
}

int main(){
    int n, m, ipt; 
    
    cin>>n;
    for(int i=0; i<n; i++){
        cin>>ipt;
        v1.push_back(ipt);
    }
    sort(v1.begin(), v1.end());
    
    cin>>m;
    for(int i=0; i<m; i++){
        cin>>ipt;
        v2.push_back(ipt);
    }
    
    for(int i=0; i<m; i++){
        if(binarySearch(v2[i])==-1) cout<< 0 << '\n';
        else cout << 1 << '\n';
    }
}